src/allotropy/parsers/beckman_pharmspec/pharmspec_parser.py

```diff
@@ -0,0 +1,151 @@
+from typing import Any
+
+import pandas as pd
+
+from allotropy.allotrope.models.light_obscuration_rec_2021_12_light_obscuration import (
+    DistributionDocumentItem,
+    DistributionItem,
+    MeasurementDocument,
+    Model,
+)
+from allotropy.allotrope.models.shared.definitions.custom import (
+    TQuantityValueCountsPerMilliliter,
+    TQuantityValueMicrometer,
+    TQuantityValueMilliliter,
+    TQuantityValueUnitless,
+)
+from allotropy.named_file_contents import NamedFileContents
+from allotropy.parsers.vendor_parser import VendorParser
+
+# This map is used to coerce the column names coming in the raw data
+# into names of the allotrope properties.
+column_map = {
+    "Cumulative Counts/mL": "cumulative particle density",
+    "Cumulative Count": "cumulative count",
+    "Particle Size(µm)": "particle size",
+    "Differential Counts/mL": "differential particle density",
+    "Differential Count": "differential count",
+}
+
+property_lookup = {
+    "particle_size": TQuantityValueMicrometer,
+    "cumulative_count": TQuantityValueUnitless,
+    "cumulative_particle_density": TQuantityValueCountsPerMilliliter,
+    "differential_particle_density": TQuantityValueCountsPerMilliliter,
+    "differential_count": TQuantityValueUnitless,
+}
+
+
+def get_property_from_sample(property_name: str, value: Any) -> Any:
+    return property_lookup[property_name](value=value)
+
+
+class PharmSpecParser(VendorParser):
+    def to_allotrope(self, named_file_contents: NamedFileContents) -> Model:
+        df = pd.read_excel(named_file_contents.contents, header=None, engine="openpyxl")
+        return self._get_model(df)
+
+    def _get_model(self, df: pd.DataFrame) -> Model:
```

Comment:
Nit: doesn't seem like you need the redirect to `_setup_model` (perhaps just rename `_setup_model` to `_get_model` and remove this?)

Reply:
Done.